src/NLog/Config/Factory.cs

```diff
@@ -146,6 +149,22 @@ public void RegisterDefinition(string itemName, Type itemDefinition)
             RegisterDefinition(itemName, itemDefinition, string.Empty, string.Empty);
         }
 
+        private string NormalizeName(string itemName)
+        {
+            if (_ignoreCharacters == null)
+            {
+                return itemName;
+            }
+
+            for (int i = 0; i < _ignoreCharacters.Count; i++)
```

Comment:
Could this be `foreach` instead ?

Reply:
sure, but AFAIK `for` was preferred over `foreach` (allocations?)

Reply:
`foreach` only allocates when using interfaces like `IEnumerable` or `IList`. Not when using actual implementation like `char[]` or `List<string>`